onnxscript/testing/benchmark/benchmark_helpers.py

```diff
@@ -0,0 +1,682 @@
+                raise BenchmarkError(
+                    f"Unable (2) to continue with config {config}, no metric was "
+                    f"collected.\n--ERROR--\n{serr}\n--OUTPUT--\n{sout}"
+                )
+            else:
+                metrics = {}
+        metrics.update(config)
+        metrics["ERROR"] = serr
+        metrics["OUTPUT"] = sout
+        metrics["CMD"] = f"[{' '.join(cmd)}]"
+        data.append(metrics)
+        if verbose > 5:
+            print("--------------- ERROR")
+            print(serr)
+        if verbose >= 10:
+            print("--------------- OUTPUT")
+            print(sout)
+
+    return data
+
+
+def common_export(
+    model: Any,
+    inputs: Sequence[Any],
+    exporter: str = "dynamo",
+    target_opset: int = 18,
+    folder: str = "",
+    filename: str = "model.onnx",
+    dynamic_shapes: Any | None = None,
+    verbose: int = 0,
+    optimization: str | None = None,
+    stats: dict[str, Any] | None = None,
+):
+    """
+    Exports a model into a folder.
+
+    Args:
+        model: model
+        exporter: script, dynamo
+        folder: folder to export into
+        filename: onnx filename
+        inputs: inputs
+        dynamic_shapes: dynamic shapes
+        target_opset: target opset
+        optimization: optimization scenario
+        verbose: verbosity
+        stats: if not None, populates this
+            dictionary with statistics about time
+
+    Returns:
+        onnx proto
+
+    """
+    import torch.onnx
+
+    if folder:
+        if not os.path.exists(folder):
+            os.mkdir(folder)
+        filename = os.path.join(folder, filename)
+
+    if verbose:
+        print(f"[common_export] start exporting with {exporter!r} in {filename!r}")
+    begin = time.perf_counter()
+    if exporter == "script":
+        torch.onnx.export(
+            model,
+            inputs,
+            filename,
+            do_constant_folding=False,
+            input_names=[f"input{i}" for i in range(len(inputs))],
+            opset_version=target_opset,
+        )
+    elif exporter == "dynamo":
+        with torch.no_grad():
+            prog = torch.onnx.dynamo_export(model, *inputs)
+        onx = prog.model_proto
+        with open(filename, "wb") as f:
+            f.write(onx.SerializeToString())
+    else:
+        raise AssertionError(f"Unknown exporter {exporter!r}")
+
+    if stats is not None:
+        stats["export_time"] = time.perf_counter() - begin
+
+    if verbose:
+        print(f"[common_export] exporter done in {time.perf_counter() - begin}s")
+        print(f"[common_export] size of the export: {os.stat(filename).st_size / 2**20} Mb")
+
+    with open(filename, "rb") as f:
+        onx = onnx.load(filename)
+
+    if optimization:
+        if verbose:
+            print(f"[common_export] start optimization with {optimization!r}")
+        begin = time.perf_counter()
+        onx = optimize_model_proto(onx, optimization, verbose=verbose, stats=stats)
```

Comment:
nit: consider more descriptive names for variables. e.g. `optimized` or `optimized_model`. `onx` tends to not reveal what the object really is.

Reply:
done